.github/workflows/format_lint_docs.yml

```diff
@@ -0,0 +1,35 @@
+name: format, lint, docs
+on: [pull_request]
+
+jobs:
+  format_lint:
```

Comment:
why not just use https://pre-commit.com/, and https://github.com/pre-commit/action or even https://pre-commit.ci/

Reply:
We can do this if it works reliably. I had troubles installing it in the past and found it an annoying overhead.
Auto-format and linting happens automatically in the IDE for me. But I won't judge other developers workflows.

Let's do this in another PR. I want to get this merged so I can work on the Sphinx/extension performance.
The current solution works and the CI prevents merging unformatted code or lint errors.

Reply:
Another thought. Let's say you have new code that you want to push to a branch. It's work in progress, with lint errors. How can I ever commit this with pre-commit hooks installed?

Reply:
`git commit -m "my message" --no-verify `` :)

I use this quite often, or nearly always at customer projects, which are using commercial tools inside their pre-commits which I don't have.

Reply:
> How can I ever commit this with pre-commit hooks installed?

it will only run automatically if you do `pre-commit install`, and you can always do `pre-commit uninstall`, otherwise you are free to push as you like

Reply:
Let's add it with another PR then.